backend/src/main/java/fk/prof/backend/model/request/RecordedProfileIndexes.java

```diff
@@ -0,0 +1,45 @@
+package fk.prof.backend.model.request;
+
+import com.koloboke.collect.map.hash.HashIntObjMap;
+import com.koloboke.collect.map.hash.HashIntObjMaps;
+import com.koloboke.collect.map.hash.HashLongObjMap;
+import com.koloboke.collect.map.hash.HashLongObjMaps;
+import recording.Recorder;
+
+import java.util.List;
+
+public class RecordedProfileIndexes {
+  private final HashLongObjMap<String> methodLookup = HashLongObjMaps.newUpdatableMap();
+  private final HashIntObjMap<String> traceLookup = HashIntObjMaps.newUpdatableMap();
+
+  public String getMethod(long methodId) {
+    return methodLookup.get(methodId);
+  }
+
+  public String getTrace(int traceId) {
+    return traceLookup.get(traceId);
+  }
+
+  public void update(Recorder.IndexedData indexedData) {
+    updateMethodIndex(indexedData.getMethodInfoList());
+    updateTraceIndex(indexedData.getTraceCtxList());
+  }
+
+  private void updateMethodIndex(List<Recorder.MethodInfo> methods) {
+    if (methods != null) {
```

Comment:
This was because I incorrectly assumed that protobuf will return null for repeated fields if no element is provided for the field. Will be removing such null checks.